### PR TITLE
Fix remote project execution

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?

Fixes currently-unreleased bug in executing projects remotely on Databricks

## How is this patch tested?

Tested manually

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/projects/__init__.py [hunk 1]
```diff
@@ -83,15 +83,17 @@ def _resolve_experiment_id(experiment_name=None, experiment_id=None):
     return _get_experiment_id()
 
 
-def _run(uri, experiment_id, entry_point="main", version=None, parameters=None,
```
Comment (smurching) [inline mlflow/projects/__init__.py:86]: Removing duplicate defaults from this internal method

File: mlflow/projects/__init__.py [hunk 2]
```diff
@@ -193,7 +195,7 @@ def _run(uri, experiment_id, entry_point="main", version=None, parameters=None,
 
 def run(uri, entry_point="main", version=None, parameters=None,
```
Comment (smurching) [inline mlflow/projects/__init__.py:196]: Adding the default `backend` arg explicitly to the public `mlflow.run` API

File: mlflow/projects/databricks.py [hunk 3]
```diff
@@ -260,8 +260,8 @@ def _get_databricks_run_cmd(dbfs_fuse_tar_uri, run_id, entry_point, parameters):
                                            posixpath.basename(dbfs_fuse_tar_uri)))
     project_dir = posixpath.join(DB_PROJECTS_BASE, tar_hash)
 
-    mlflow_run_cmd = " ".join(_get_cluster_mlflow_run_cmd(
-        project_dir, run_id, entry_point, parameters))
+    mlflow_run_arr = _get_cluster_mlflow_run_cmd(project_dir, run_id, entry_point, parameters)
+    mlflow_run_cmd = " ".join([shlex_quote(elem) for elem in mlflow_run_arr])
```
Comment (smurching) [inline mlflow/projects/databricks.py:264]: This is the underlying bug fix, we need to string-escape the individual args in the command before joining them together as a raw string to pass to `bash -c` below

**Review by tomasatdatabricks — APPROVED** [review]
lgtm!

**Pull request merged**